benches/lib.rs

```diff
@@ -1,8 +1,8 @@
 use criterion::{criterion_group, criterion_main, Criterion};
 use oxiri::Iri;
 
-fn bench_iri_parse(c: &mut Criterion) {
-    let examples = [
+fn abs_examples() -> &'static [&'static str] {
```

Comment:
Why not just using a `const` like `const abs_example: [&str; 24] = `?

Reply:
a function is easier to maintain, as you don't have to hardcode the number of examples... but yes, a `const` would work

Reply:
> What about having all the code inside of the `IriRef` type and have the `Iri` type as a frontend to this type that ensures the invariant "the IRI is absolute".

yes, good idea.